File: app/src/main/java/fr/everydaysapps/marvelsuperheroes/screens/MainActivity.java
package fr.everydaysapps.marvelsuperheroes.screens;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.WindowManager;

import fr.everydaysapps.marvelsuperheroes.R;
import fr.everydaysapps.marvelsuperheroes.screens.heroes.HeroesListActivity;
import fr.everydaysapps.marvelsuperheroes.utils.UiUtils;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        UiUtils.launchFragmentKeepingInBackStack(this, new HeroesListActivity());
    }

}

File: app/src/main/java/fr/everydaysapps/marvelsuperheroes/ui/MainActivity.kt
package fr.everydaysapps.marvelsuperheroes.ui

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.compose.rememberNavController
import dagger.hilt.android.AndroidEntryPoint
import fr.everydaysapps.marvelsuperheroes.ui.theme.MarvelSuperHeroesTheme

@AndroidEntryPoint
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MarvelSuperHeroesTheme {
                val navController = rememberNavController()
                NavGraph(navController = navController)
            }
        }
    }
}
